pkg/controller/template/render.go

```diff
@@ -294,53 +293,31 @@ func skipMissing(key string) (interface{}, error) {
 
 // Process the {{etcdPeerCertDNSNames}} and {{etcdServerCertDNSNames}}
 func etcdServerCertDNSNames(cfg renderConfig) (interface{}, error) {
-	if cfg.ClusterName == "" || cfg.BaseDomain == "" || cfg.EtcdInitialCount <= 0 {
+	if cfg.BaseDomain == "" {
 		return nil, fmt.Errorf("invalid configuration")
 	}
 
 	var dnsNames = []string{
 		"localhost",
-		"*.kube-etcd.kube-system.svc.cluster.local",
-		"kube-etcd-client.kube-system.svc.cluster.local",
 		"etcd.kube-system.svc",               // sign for the local etcd service name that cluster-network apiservers use to communicate
 		"etcd.kube-system.svc.cluster.local", // sign for the local etcd service name that cluster-network apiservers use to communicate
-	}
-
-	for i := 0; i < cfg.EtcdInitialCount; i++ {
-		dnsNames = append(dnsNames, fmt.Sprintf("%s-etcd-%d.%s", cfg.ClusterName, i, cfg.BaseDomain))
+		"${ETCD_DNS_NAME}",
 	}
 	return strings.Join(dnsNames, ","), nil
 }
 
 func etcdPeerCertDNSNames(cfg renderConfig) (interface{}, error) {
-	if cfg.ClusterName == "" || cfg.BaseDomain == "" || cfg.EtcdInitialCount <= 0 {
+	if cfg.BaseDomain == "" {
 		return nil, fmt.Errorf("invalid configuration")
 	}
 
 	var dnsNames = []string{
-		"*.kube-etcd.kube-system.svc.cluster.local",
-		"kube-etcd-client.kube-system.svc.cluster.local",
-	}
-
-	for i := 0; i < cfg.EtcdInitialCount; i++ {
-		dnsNames = append(dnsNames, fmt.Sprintf("%s-etcd-%d.%s", cfg.ClusterName, i, cfg.BaseDomain))
+		"${ETCD_DNS_NAME}",
+		cfg.BaseDomain, // https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/clustering.md#dns-discovery
```

Comment:
Instead of master, use a specific commit so that document doesn't disappear on us.